### PR TITLE
bevy_color: Add Tailwind palette

### PR DESCRIPTION
**Pull request by rparrett**
# Objective

Give Bevy a well-designed built-in color palette for users to use while prototyping or authoring Bevy examples.

## Solution

Generate ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f7b3a3002fb7727db15c1197e0a1a373), [gist](https://gist.github.com/rust-play/f7b3a3002fb7727db15c1197e0a1a373)) consts from [Tailwind](https://tailwindcss.com/docs/customizing-colors) (mit license) json.

## Discussion

Are there other popular alternatives we should be looking at? Something new and fancy involving a really long acronym like CIELUVLCh? I'm not a tailwind user or color expert, but I really like the way it's broken up into distinct but plentiful hue and lightness groups.

It beats needing some shades of red, scrolling through the [current palette](https://docs.rs/bevy/latest/bevy/prelude/enum.Color.html), choosing a few of `CRIMSON`, `MAROON`, `RED`, `TOMATO` at random and calling it a day.

The best information I was able to dig up about the Tailwind palette is from this thread: https://twitter.com/steveschoger/status/1303795136703410180. Here are some key excerpts:

> Tried to the "perceptually uniform" thing for Tailwind UI. 
> Ultimately, it just resulted in a bunch of useless shades for colors like yellow and green that are inherently brighter. 

>  With that said you're guaranteed to get a contrast ratio of 4.5:1 when using any 700 shade (in some cases 600) on a 100 shade of the same hue.

> We just spent a lot of time looking at sites to figure out which colors are popular and tried to fill all the gaps. 
> Even the lime green is questionable but felt there needed to be something in between the jump from yellow to green 😅

### CODE REVIEW

**Comment by IceSentry** [comment]
I'm pretty we should include the tailwind license in the file and maybe also mention it in the docs

**Comment by IceSentry** [comment]
Here's how I did that in an old crate of mine https://github.com/IceSentry/bevy_ui_styled/blob/master/src/colors.rs

**Review by IceSentry — APPROVED** [review]
I'll approve, but I think we'll want @cart involved here because of the license situation. It's a bit of an edge case if someone wants to use bevy with the apache license they can't use that module I guess. I honestly don't know enough about licensing.

File: crates/bevy_color/src/palettes/tailwind.rs [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+/// <div style="background-color:rgb(251, 191, 36); width: 10px; padding: 10px; border: 1px solid;"></div>
+pub const AMBER400: Srgba = Srgba::rgb(0.9843137, 0.7490196, 0.14117648);
+/// <div style="background-color:rgb(245, 158, 11); width: 10px; padding: 10px; border: 1px solid;"></div>
+pub const AMBER500: Srgba = Srgba::rgb(0.9607843, 0.61960787, 0.043137256);
+/// <div style="background-color:rgb(217, 119, 6); width: 10px; padding: 10px; border: 1px solid;"></div>
+pub const AMBER600: Srgba = Srgba::rgb(0.8509804, 0.46666667, 0.023529412);
+/// <div style="background-color:rgb(180, 83, 9); width: 10px; padding: 10px; border: 1px solid;"></div>
+pub const AMBER700: Srgba = Srgba::rgb(0.7058824, 0.3254902, 0.03529412);
+/// <div style="background-color:rgb(146, 64, 14); width: 10px; padding: 10px; border: 1px solid;"></div>
+pub const AMBER800: Srgba = Srgba::rgb(0.57254905, 0.2509804, 0.05490196);
```
Comment (JMS55) [inline crates/bevy_color/src/palettes/tailwind.rs:47]: I'd prefer e.g. AMBER_800 personally.

**Review by afonsolage — APPROVED** [review]

File: crates/bevy_color/src/palettes/tailwind.rs (hunk 1 above)
Comment (IceSentry) [reply crates/bevy_color/src/palettes/tailwind.rs:47]: Oh, yeah, I think I'd prefer that too

File: crates/bevy_color/src/palettes/tailwind.rs [hunk 2]
```diff
@@ -0,0 +1,534 @@
+//! Colors from [Tailwind CSS](https://tailwindcss.com/docs/customizing-colors) (MIT License).
```
[the remark this answers is not in the document]
Comment (rparrett) [reply crates/bevy_color/src/palettes/tailwind.rs:1]: Good idea.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
I'm pretty firmly of the opinion that this is fine from a licensing perspective. MIT allows users to relicense the code (as long as they provide attribution and a copy of the license). It doesn't include a patent grant, but this is a color palette: I don't think that's a relevant concern.

**Review by JMS55 — APPROVED** [review]

**Review by afonsolage — APPROVED** [review]

**Comment by soqb** [comment]
don't want to hamper any progress here, but couldn't this very easily be a standalone community crate?

this is of course true for the other palettes, but at least the css and stand-alone modules are more "ubiquitous".

the actual implementation is solid, though.

**Comment by alice-i-cecile** [comment]
> don't want to hamper any progress here, but couldn't this very easily be a standalone community crate?

It definitely could, but I'd like to have access to at least one "good" color palette: it's really painful trying to build nice examples and dev tools with only hand-authored magic values and the terrible CSS colors.

**Comment by rparrett** [comment]
> don't want to hamper any progress here, but couldn't this very easily be a standalone community crate?
> this is of course true for the other palettes, but at least the css and stand-alone modules are more "ubiquitous".

Definitely. There are a few existing examples I know of:
https://github.com/guimcaballero/bevy_contrib_colors
https://lib.rs/crates/bevy_ui_styled

Which I am choosing to interpret as a sign that there is demand for this. IMO, the CSS palette we do provide may be ubiquitous, but it is also next to useless and is actively harming the look and feel of our examples and our users' projects. A bit dramatic and I would still be very open to providing a single better "bevy palette" instead.

**Comment by soqb** [comment]
yep. you've convinced me.

**Comment by viridia** [comment]
I want to point out that it's pretty easy to algorithmically generate a good-looking palette, similar to Tailwind, Material Design, or whatever framework *du jour* you prefer, especially if you use Oklch to do the calculations, which gives you consistent lightness across different hues.

**Comment by viridia** [comment]
BTW, I think the tailwind palette is fine. Yes, there are other palettes - MaterialUI, NextUI, Bootstrap, etc., but they are all about the same. We don't need more than one, and people know the name Tailwind, it has brand recognition.

**Review by IceSentry — APPROVED** [review]

**Pull request merged**